hack/tools/tilt-prepare/main.go

```diff
@@ -249,19 +261,76 @@ func tiltResources(ctx context.Context, ts *tiltSettings) error {
 	}
 
 	// Add a provider task for each name/path defined using the --provider flag.
-	for _, p := range *providersBuildsFlag {
-		pValues := strings.Split(p, ":")
-		if len(pValues) != 2 {
-			return errors.Errorf("[resources] failed to parse --provider flag %s: value should be in the form of name:path", p)
+	for _, p := range ts.ProviderBuilds {
+		name, path := providerContext(p)
+		tasks[name] = providerTask(name, fmt.Sprintf("%s/config/default", path), ts)
+	}
+
+	// Add a task for the provider repos defined by the provider_repo key in tilt file
+	for _, p := range ts.ProviderRepos {
```

Comment:
This changes the behavior. I would prefer keeping it the same, i.e.
* enable_providers decides which provider is enabled or not
* provider_repos is used to retrieve the tilt-provider file 

Can we please (similar to the tiltsettings file) first build up the information for the builtin providers + the one from provider repos and then only add providerTask for the ones that are enabled?

Also core is always enabled

Reply:
A suggestion to replace l.263-l.276
```go
	providerPaths := map[string]string{
		"core":                  ".",
		"kubeadm-bootstrap":     "bootstrap/kubeadm",
		"kubeadm-control-plane": "controlplane/kubeadm",
		"docker":                "test/infrastructure/docker",
	}
	for _, p := range ts.ProviderRepos {
		name, path, err := loadProviders(p)
		if err != nil {
			return err
		}
		providerPaths[name] = path
	}

	ts.EnabledProviders = append(ts.EnabledProviders, "core")

	// Add a provider task for each enabled provider.
	for _, providerName := range ts.EnabledProviders {
		path, ok := providerPaths[providerName]
		if !ok {
			return errors.Errorf("failed to find path for provider %s", providerName)
		}
		tasks[providerName] = providerTask(providerName, fmt.Sprintf("%s/config/default", path), ts)
	}
```
(+ we can drop providerContext)